uftrace.c

```diff
@@ -694,6 +696,10 @@ static error_t parse_option(int key, char *arg, struct argp_state *state)
 		opts->diff_policy = arg;
 		break;
 
+	case OPT_format:
+		opts->format = arg;
```

Comment:
You'd better checking validity of the argument here (or somewhere in the report command).  So that we can show a warning for unsupported output formats.

Reply:
@gy741 Please check if `opts->format` is a valid type.

Reply:
Sorry I'm late, @namhyung  @honggyukim 

 How about this code?

```diff
diff --git a/cmds/report.c b/cmds/report.c
index a81da017..37d5f0b4 100644
--- a/cmds/report.c
+++ b/cmds/report.c
@@ -270,7 +270,7 @@ static void report_functions(struct uftrace_data *handle, struct opts *opts)

                print_and_delete(&sort_root, true, NULL, print_csv_function);
        }
-       else {
+       else if (opts->format && !strcmp(opts->format, "simple")) {
                if (avg_mode == AVG_NONE)
                        pr_out(f_format, "Total time", "Self time", "Calls", maxlen, "Function");
                else if (avg_mode == AVG_TOTAL)
@@ -281,6 +281,8 @@ static void report_functions(struct uftrace_data *handle, struct opts *opts)
                pr_out(f_format, line, line, line, maxlen, line);
                print_and_delete(&sort_root, true, NULL, print_function);
        }
+       else
+               pr_warn("Invalid style specified: %s: \n", opts->format);
 }

 static void add_remaining_task_fstack(struct uftrace_data *handle,
```


output:
```
root@goorm:/workspace/karas/uftrace/build/bin(add_support_report_csv)# ./uftrace report --format=csv
Total time, Self time, Calls, Function
2161, 319, 1, main
1842, 310, 1, a
1532, 379, 1, b
1153, 580, 1, c
643, 643, 1, __monstartup
573, 573, 1, getpid
220, 220, 1, __cxa_atexit

root@goorm:/workspace/karas/uftrace/build/bin(add_support_report_csv)# ./uftrace report --format=simple
  Total time   Self time       Calls  Function
  ==========  ==========  ==========  ====================
    2.161 us    0.319 us           1  main
    1.842 us    0.310 us           1  a
    1.532 us    0.379 us           1  b
    1.153 us    0.580 us           1  c
    0.643 us    0.643 us           1  __monstartup
    0.573 us    0.573 us           1  getpid
    0.220 us    0.220 us           1  __cxa_atexit

root@goorm:/workspace/karas/uftrace/build/bin(add_support_report_csv)# ./uftrace report --format=test
WARN: Invalid style specified: test:
```